src/main/java/bio/terra/service/configuration/ConfigFault.java

```diff
@@ -0,0 +1,79 @@
+package bio.terra.service.configuration;
+
+import bio.terra.app.controller.exception.ValidationException;
+import bio.terra.model.ConfigFaultModel;
+import bio.terra.model.ConfigModel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// Sub-classes must implement:
```

Comment:
I think sub-classes will have to implement all abstract methods of this class, so also testInsertFault() and resetFaultConfig(). If you want to make those optional, I think you have to make them not abstract and throw a NotImplementedException or something like that.

Reply:
I removed the obsolete comment. Probably best to let the code say it.